collect/cache/span_cache_basic.go

```diff
@@ -72,32 +70,48 @@ func (sc *spanCache_basic) Get(traceID string) *types.Trace {
 }
 
 func (sc *spanCache_basic) GetOldest(fract float64) []string {
+	type tidWithImpact struct {
+		id     string
+		impact int
+	}
 	n := int(float64(len(sc.cache)) * fract)
```

Comment:
len(sc.cache) -> sc.Len()?

Reply:
I get what you're going for, and it's technically safe here, but we have locking going on here, and using the public (locked) Len function in a routine that also locks is dangerous -- it could cause a deadlock if someone rearranges code. We could have an unlocked helper function that's used by both, but for something like this it doesn't seem worth it to me.

Reply:
Right. ~~My main concerns is the race condition here. Let's say the a call to `Remove(traceIDs)` happened right after the `len(sc.cache)` here. Now we may end up with a `n` that's bigger than the number of items in the new `sc.cache`.~~

I missed the check on line 98. Ok, I don't think it matters here that we get an accurate n now